src/Components/App.js

```diff
@@ -150,122 +148,77 @@ class App extends Component {
 
-    switch (dataId) {
-
-      case 'card':
-        if (document.querySelector('#card').checked) {
-          holder.type.card = true;
-          holder.type.board = false;
-        } else {
-          holder.type.card = false
-        }
-        document.querySelector('#board').checked = false
-        break;
-      case 'board':
-        if (document.querySelector('#board').checked) {
-          holder.type.board = true;
-          holder.type.card = false;
-        } else {
-          holder.type.board = false
-        }
-        document.querySelector('#card').checked = false
-        break;
-
-      case 'two':
-        (document.querySelector('#two').checked) ? holder.players.two = true : holder.players.two = false;
-        break;
-      case 'threefour':
-        (document.querySelector('#threefour').checked) ? holder.players.threefour = true : holder.players.threefour = false;
-        break;
-      case 'fivesix':
-        (document.querySelector('#fivesix').checked) ? holder.players.fivesix = true : holder.players.fivesix = false;
-        break;
-      case 'seven':
-        (document.querySelector('#seven').checked) ? holder.players.seven = true : holder.players.seven = false;
-        break;
-
-      case 'lesseight':
-        (document.querySelector('#lesseight').checked) ? holder.age.lesseight = true : holder.age.lesseight = false;
-        break;
-      case 'eightthirteen':
-        (document.querySelector('#eightthirteen').checked) ? holder.age.eightthirteen = true : holder.age.eightthirteen = false;
-        break;
-      case 'thirteenplus':
-        (document.querySelector('#thirteenplus').checked) ? holder.age.thirteenplus = true : holder.age.thirteenplus = false;
-        break;
-      case 'adult':
-        (document.querySelector('#adult').checked) ? holder.age.adult = true : holder.age.adult = false;
-        break;
+    if (dataCategory && document.querySelector(`#${dataFilter}`).checked) {
+      filterOptions[dataCategory][dataFilter] = true
+    } else if (dataCategory) {
+      filterOptions[dataCategory][dataFilter] = false;
+    }
 
-      case 'strategy':
-        (document.querySelector('#strategy').checked) ? holder.genre.strategy = true : holder.genre.strategy = false;
-        break;
-      case 'family':
-        (document.querySelector('#family').checked) ? holder.genre.family = true : holder.genre.family = false;
-        break;
-      case 'party':
-        (document.querySelector('#party').checked) ? holder.genre.party = true : holder.genre.party = false;
-        break;
-      case 'adventure':
-        (document.querySelector('#adventure').checked) ? holder.genre.adventure = true : holder.genre.adventure = false;
-        break;
+    if (dataFilter === 'card') {
+      filterOptions.type.board = false
+      document.querySelector('#board').checked = false
+    } else if (dataFilter === 'board') {
+      filterOptions.type.card = false;
+      document.querySelector('#card').checked = false
```

Comment:
This is for the 2 fringe cases of card and board since they have to act as a radio button that can both be off too